api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperties.java

```diff
@@ -65,7 +64,6 @@
      *
      * @return the configuration property prefix
      */
-    @Nonbinding
```

Comment:
Can you explain why you need to delete this? Once you delete, you will force impl to create different beans with different prefix.

Reply:
I think not.

The issue is that with @Nonbinding, the prefix value is not taken into account when matching Qualifiers, so if you have:

```java
@ConfigProperties(prefix="foo")
SomeProperties propertiesFoo;

@ConfigProperties
SomeProperties properties;

@ConfigProperties
class SomeProperties {
}
```

You would create 2 instances beans for `SomeProperties`, one using the `@ConfigProperties` qualifier and the other 
the `@ConfigProperties(prefix="foo")` qualifier. In this situation you get an ambiguous resolution.

The prefix does identify the config namespace we want to use associated with the bean. It seems to me that the `prefix` should be binded to the qualifier as well. What do you think?

Reply:
This depends on the implementation. If you ask for adding `@NonBinding`, you force implementations to provide multiple beans for each prefix used in the code. If you don't add `@NonBinding`, the implementation can just provide one bean and give out different instance depending on the prefix. I think it is much more flexible. Besides, the name attribute is also `NonBinding` in the `ConfigProperty`.

Reply:
I think it makes some sense.

It the case of `ConfigProperty` we are talking about a potentially high number of combinations, so it doesn't make sense to provide a bean for each.

In the case of `ConfigProperties`, the number will be much lower, plus it does simplify the implementation and possible extensions, because consumers have to check the injection point prefix, qualifier prefix and finally annotated type prefix when providing the new instance to get the right configuration.

Reply:
How do you create CDI beans? Do you base on the config properties? Do you add beans based on the injection points?  It will be helpful to get some insights from other implementations as well. @jbee thoughts?

Reply:
I do both. I collect the annotated classes first and provide beans for them. Then I also look into the injection points and provide beans for them.

And completely unrelated, we may want to revisit ao we do injection of a `ConfigProperties` class. Right now, we only need to use `@Inject`, unless we want to override the prefix and it is fine.

On the other hand, after trying it for a while, I feel that adding a `@ConfigProperties` all the time at the injection points, makes the code more clear, in the sense that I can tell straight away what type of class that is. Also, this made me question if we should override the injection point with an empty prefix. Right now, adding a simple `@ConfigProperties` will always override the prefix to use an empty one.

Reply:
I'll open a separate issue so we can discuss a few things. Right now, I think we may have too many options.